parser/old_parser_components/parser.py

```diff
@@ -60,7 +67,13 @@ def _parse(location, prefixes, validate_version):
             # Check for no node templates
             if (context.modeling.model) and (not context.modeling.model.node_templates):
                 context.validation.report('no node templates')
-        
+
     context.validation.dump_issues()
 
-    return context
+    if context.validation.has_issues:
+        msg = 'issues:\n'
```

Comment:
1) Issues
2) use StringIO and not msg +=
3) maybe refactor / use aria/parser/validation/context.py:ValidationContext.dump_issues()